### PR TITLE
Crash: Polymorphic field in a within With() expression 

### PR DESCRIPTION
**Pull request by jas-valgotar**
-Fixes https://github.com/microsoft/Power-Fx-Dataverse/issues/171

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs [hunk 1]
```diff
@@ -290,7 +290,7 @@ public override async ValueTask<FormulaValue> Visit(CallNode node, EvalVisitorCo
                             });
                     }
                     
-                    if (!(result.IRContext.ResultType == node.IRContext.ResultType || result is ErrorValue || result.IRContext.ResultType is BlankType || func is IfErrorFunction))
+                    if (!(result.IRContext.ResultType._type == node.IRContext.ResultType._type || result is ErrorValue || result.IRContext.ResultType is BlankType || func is IfErrorFunction))
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs:293]: >_type [](http://example.com/codeflow?start=54&length=5)

Do we need this? Looks like it's to invoke Dtype's equality. 
FormulaType overrides == already and forward to Dtype?

**Review by MikeStall — DISMISSED** [review]

File: src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs (hunk 1 above)
Comment (jas-valgotar) [reply src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs:293]: Yes, we should only rely on DType inside Binder...

Think of 2 different implementation of RecordType, If they have same underlying dtype, should this check pass?

File: src/libraries/Microsoft.PowerFx.Core/Public/Types/RecordType.cs [hunk 2]
```diff
@@ -99,6 +99,11 @@ public static RecordType Empty()
             return new KnownRecordType();
         }
 
+        public static RecordType Polymorphic()
+        {
+            return new KnownRecordType(DType.Polymorphic);
```
[the remark this answers is not in the document]
Comment (jas-valgotar) [reply src/libraries/Microsoft.PowerFx.Core/Public/Types/RecordType.cs:104]: We should loose it to have compile and runtime type compat.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by LucGenetier — APPROVED** [review]
:shipit:

**Pull request merged**